segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -418,7 +420,7 @@ public boolean isOffline() {
     public CompletableFuture<Void> createStreamSegment(String streamSegmentName, Collection<AttributeUpdate> attributes, Duration timeout) {
         ensureRunning();
 
-        logRequest("createStreamSegment", streamSegmentName);
+        logTrackedRequest("createSegment", "createStreamSegment", streamSegmentName);
```

Comment:
I think for the Segment Store side, it is sufficient to stop at PravegaRequestProcessor and AppendProcessor. No need to go down into `StreamSegmentContainer` or Tier2. 

I would suggest rolling back changes to this class and to `RollingStorage` and just keeping the request tracing at the highest layer. As you correctly mentioned in the PDP, you have no guarantees as to where your request is headed; it might just go to the metadata or directly to Tier2. Let's not make any more hardcoded assumptions here, and keep this decoupled from underlying layers. The Segment store logs already have enough correlation to be able to track requests from here on.

Reply:
I accepted your suggestion and removed the request tracking in Tier2/segment container. Anyway, I personally found it very useful to do a `grep` and find all the activity related to a client's request across multiple logs, including Tier 2.

Reply:
If the trace id we are adding here is not propagated to the segment store (i.e., we treat the segment store as a black box), then how do we correlate a particular request with the processing inside the segment store? One of the goals of this work, as @RaulGracia is pointing out, is to give us the ability to easily grep over all logs to find the activity triggered by a given request. If we stop at the request processor, then we will need to somehow correlate the request with the segment store processing for debugging purposes. 

Please clarify this point. @andreipaduroiu @RaulGracia

Reply:
Once in the Segment Store all segment-related log entries (where applicable) are tagged with either the Segment Id or Segment Name (and somewhere in the logs we log that correlation too). 

IMO, it makes no sense to percolate this down into the Store because the definition of external request quickly loses its meaning. Requests are converted into Operations and put in a queue. Some other thread picks them up, packages them into frames, and writes them to tier1. Some other thread picks up the Operations, regroups them, and ships them to Tier2. 

As soon as we reach that queue, there is no long a "request"; it's an internal task.

We could go out of our ways to propagage this information down to every single component, but just imagine the amount of plumbing and bookkeeping we'd have to do just for this.

And what happens after recovery? If a request (which becomes an operation) is recovered from Tier1 on a different machine, then all these request contexts will not exist anymore, so we'd be stuck not writing anything of value there.